### PR TITLE
Ensure compliance with return types in APE 14 API

### PR DESCRIPTION
**Pull request by Cadair**

### CODE REVIEW

File: gwcs/api.py [hunk 1]
```diff
@@ -170,7 +175,25 @@ def pixel_bounds(self):
         WCS that includes fitted distortions. This is an optional property,
         and it should return `None` if a shape is not known or relevant.
         """
-        return self.bounding_box
+        bounding_box = self.bounding_box
+        if bounding_box is None:
+            return bounding_box
+
+        if self.pixel_n_dim == 1 and len(bounding_box) == 2:
+            bounding_box = (bounding_box,)
```
[the remark this answers is not in the document]
Comment (Cadair) [reply gwcs/api.py]: I am pretty sure this failed the test I wrote.

[the remark this answers is not in the document]
Comment (nden) [reply gwcs/api.py]: Ah, I read this backwards. Yes, for 1D `bounding_box` always returns a tuple of 2 floats.
So `pixel_to_world` returns one output in this case but `pixel_bounds` returns an iterable? 
I wonder if this makes it more difficult for applications to work with `pixel_bounds` as they have to check whether `n_dim` is 1 and retrieve the 0-th element.

[the remark this answers is not in the document]
Comment (Cadair) [reply gwcs/api.py]: I see your point, this is probably worth raising on my type hint PR on astropy core.

**Comment by nden** [comment]
This PR depends on astropy/astropy#9677. The testing of the current changes is skipped if code in 
 astropy/astropy#9677 is not available. Seemingly tests pass here, however if astropy/astropy#9677 is installed there are 10 failures.

**Comment by Cadair** [comment]
@nden I think I have fixed the test build with the type checking here (based on this branch and the latest commits to astropy/astropy#9677) If we can get this merged I think it fixes @nmearl / @eteq 's issues in specutils, so they can work on testing their fixes while I polish off the other open APE 14 PRs using this as a base. (I have run into type issues fixed in here in the stokes PR for instance).

**Comment by nden** [comment]
Looks good. Thanks @Cadair !

**Pull request merged**